packages/lisk-transactions/src/utils/sign_and_verify.ts

```diff
@@ -36,8 +45,109 @@ export const multiSignTransaction = (
 	return cryptography.signData(transactionHash, passphrase);
 };
 
+interface VerifySignatureReturn {
+	readonly verified: boolean;
+	readonly error?: TransactionError;
+}
+
+export const verifySignature = (
+	publicKey: string,
+	signature: string,
+	transactionBytes: Buffer,
+	id?: string,
+): VerifySignatureReturn => {
+	const transactionHash = cryptography.hash(transactionBytes);
+
+	const verified = cryptography.verifyData(
+		transactionHash,
+		signature,
+		publicKey,
+	);
+
+	return {
+		verified,
+		error: !verified
+			? new TransactionError(
+					`Failed to verify signature ${signature}`,
+					id,
+					'.signature',
+			  )
+			: undefined,
+	};
+};
+
+interface VerifyMultisignatureReturn {
+	readonly verified: boolean;
+	readonly errors: ReadonlyArray<TransactionError>;
+}
+
+export const verifyMultisignatures = (
+	memberPublicKeys: ReadonlyArray<string> = [],
+	signatures: ReadonlyArray<string>,
+	minimumValidations: number,
+	transactionBytes: Buffer,
+	id?: string,
+): VerifyMultisignatureReturn => {
+	if (!minimumValidations || typeof minimumValidations !== 'number') {
+		return {
+			verified: false,
+			errors: [new TransactionError('Sender does not have valid multimin', id)],
+		};
+	}
+	if (!signatures || (signatures && signatures.length < minimumValidations)) {
+		return {
+			verified: false,
+			errors: [new TransactionError('Missing signatures', id, '.signatures')],
+		};
+	}
+
+	const verifiedSignatures: string[] = [];
+	const checkedPublicKeys: string[] = [];
+
+	memberPublicKeys.forEach(publicKey => {
+		if (checkedPublicKeys.includes(publicKey)) {
+			return;
```

Comment:
this condition will never be true unless memberPublicKeys are not unique

Reply:
it can be true as per [issue #2540](https://github.com/LiskHQ/lisk/issues/2540). in core logic this check is also done in the same way.